WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -96,6 +100,14 @@ def index
       @competitions = @competitions.select { |competition| competition.has_events_with_ids?(params[:event_ids]) }
     end
 
+    unless params[:status] == "all"
+      if params[:status] == "warning"
+        @competitions = @competitions.select { |competition| competition.pending_results_or_report(14) }
+      else
+        @competitions = @competitions.select { |competition| competition.pending_results_or_report(21) }
```

Comment:
I agree with @larspetrus, there is no need to call `.select` in 2 different places.

We still need to address the 3 occurrences of the numbers 14 and 21 in our code base, though.